hips/hip-9999.md

```diff
@@ -198,6 +203,7 @@ The distribution format may either be discrete files (ie. filesystem or version
 
 Eventually, it is recommended that Helm supports plugin distribution via OCI artifacts. And further distribtion schemes e.g. S3 could be supported via plugins themselves.
 
+To enable airgapped installations and other use cases where downloading Chart-defined plugins is not feasible, plugins may be packaged and distributed with the chart itself, similar to how Helm 3 allows this for a chart's subchart dependencies. A command will be added to enable this (see [Helm plugin commands](#Helm-plugin-commands)).
```

Comment:
The distribution via OCI and mirroring / aliasing of OCI registries would also go a long way to solving for air-gaped environments.

Bundling plugins with charts has the downside that charts will become much larger (go template render is 19MiB uncompressed / ~3MiB compressed for example). But otherwise bundling is completely possible of course.